logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -111,7 +70,59 @@ def config_name
   end
 
   def register
-    @workers.each {|w| w.register}
+    raise ArgumentError, "Attempted to register #{self} twice!" if @registered
+    @registered = true
+    # We define this as an array regardless of threadsafety
+    # to make reporting simpler, even though a threadsafe plugin will just have
+    # a single instance
+    #
+    # Older plugins invoke the instance method Outputs::Base#workers_not_supported
+    # To detect these we need an instance to be created first :()
+    # TODO: In the next major version after 2.x remove support for this
+    @workers << @klass.new(@config)
+    @workers.first.register # Needed in case register calls `workers_not_supported`
```

Comment:
not sure I understand why we call register only to the first element of @workers, but @workers is a global array and this methods can only be called once. isn't this register call not going to happen always regarder of the way the output plugin works?